OrangeHRMTestFramework/PageObjects/OrangeHRM/Filters/UserManagementFilter.cs

```diff
@@ -0,0 +1,43 @@
+﻿using OpenQA.Selenium;
+using OrangeHRMTestFramework.Common.WebElements;
+using OrangeHRMTestFramework.Data.Constants;
+using OrangeHRMTestFramework.Data.Enums;
+
+namespace OrangeHRMTestFramework.PageObjects.OrangeHRM.Filters
+{
+    public class UserManagementFilter: BaseFilter
+    {
+        public void EnterValueToUserNameFilterInput(string value) => new OrangeWebElement(By.XPath(string.Format(BaseFilterInputLocator, UserManagementFieldNames.Username))).SendKeys(value);
+
+        public void SelectValueInStatusDropdown<TEnum>(TEnum value) where TEnum : struct, IConvertible, IComparable, IFormattable
+        {
+            SelectValueInDropdown(UserManagementFieldNames.Status, value);
+        }
+
+        public void SelectValueInUserRoleDropdown<TEnum>(TEnum value) where TEnum : struct, IConvertible, IComparable, IFormattable
+        {
+            SelectValueInDropdown(UserManagementFieldNames.UserRole, value);
+        }
+
+        public void EnterValuesToAllFiltersAndApply(string valueForUserName, string valueForEmployeeName, Status valueOfStatus, UserRole valueOfUserRole)
+        {
+                EnterValueToUserNameFilterInput(valueForUserName);
+                EnterAndSelectValueInEmployeeNameFilterInput(valueForEmployeeName);
+                SelectValueInStatusDropdown(valueOfStatus);
+                SelectValueInUserRoleDropdown(valueOfUserRole);
+                ClickSearchButton();
+        }
+
+        private void SelectValueInDropdown<TEnum>(string fieldName, TEnum value) where TEnum : struct, IConvertible, IComparable, IFormattable
```

Comment:
Maybe I'm wrong. But it looks a bit complicated. Why don't you take in this method string value. So in public method you fill pass an enum value with ToString() and public methods will take specific Enums

Reply:
Please check it in other classes

Reply:
Good point 🥲  Fixed.